src/vtl.ts

```diff
@@ -526,6 +537,32 @@ export abstract class VTL {
     } else if (isDoStmt(node)) {
     } else if (isTypeOfExpr(node)) {
     } else if (isWhileStmt(node)) {
+    } else if (isAwaitExpr(node)) {
+      // we will check for awaits on the PromiseExpr
+      return this.eval(node.expr);
+    } else if (isPromiseExpr(node)) {
+      // if we find a promise, ensure it is wrapped in Await or returned then unwrap it
+      if (isAwaitExpr(node.parent) || isReturnStmt(node.parent)) {
+        return this.eval(node.expr);
+      }
+      debugger;
+      throw new SynthError(
+        ErrorCodes.Integration_must_be_immediately_awaited_or_returned
+      );
+    } else if (isPromiseArrayExpr(node)) {
+      // if we find a promise array, ensure it is wrapped in a Promise.all then unwrap it
+      if (
+        isArgument(node.parent) &&
+        isCallExpr(node.parent.parent) &&
+        isPromiseAll(node.parent.parent)
+      ) {
+        return this.eval(node.expr);
+      }
+      debugger;
+      // TODO create error code
+      throw new SynthError(
+        ErrorCodes.Arrays_of_Integration_must_be_immediately_wrapped_in_Promise_all
+      );
```

Comment:
What does it mean to do parallel promises in appsync?

Reply:
Think we should just fail if a `map` returns a promise or uses one?

Reply:
PromiseAll is evaluated above

Reply:
Yeah, messed up here, there are two cases. 

1. Must wrap PromiseArrays with Promise.all
2. Must only use Promise.all to wrap PromiseArrays

Fixing